source/Hardpoint.cpp

```diff
@@ -407,3 +471,54 @@ void Hardpoint::Fire(Ship &ship, const Point &start, const Angle &aim)
 	// case the outfit is its own ammunition.
 	ship.ExpendAmmo(*outfit);
 }
+
+
+
+// The arc depends on both the base hardpoint and the installed outfit.
+void Hardpoint::UpdateArc()
+{
+	if(!outfit)
+		return;
+
+	const BaseAttributes &attributes = baseAttributes;
```

Comment:
Why?

Reply:
Not sure, I didn't write this part. For the `const` maybe?

Reply:
tbh baseAttributes could be made const themselves

Reply:
Fixed in last commit